test/addons_test.go

```diff
@@ -214,6 +214,61 @@ func TestAzureGroup(t *testing.T) {
 // Private Functions
 // -----------------------------------------------------------------------------
 
+func checkIfUpgradeIsNeeded(t *testing.T, groupname string) (bool, []v1beta2.AddonInterface, error) {
+	t.Logf("determining which addons in group %s need to be upgrade tested", groupname)
+
+	var doUpgrade bool
+	addons := groups[groupname]
+	for _, addon := range addons {
+		if err := overrides(addon); err != nil {
+			return false, nil, err
+		}
+	}
+
+	addonDeploymentsArray := []v1beta2.AddonInterface{}
+	for _, newAddon := range addons {
+		t.Logf("verifying whether upgrade testing is needed for addon %s", newAddon.GetName())
+		oldAddon, err := addontesters.GetLatestAddonRevisionFromLocalRepoBranch("../", comRepoRemote, comRepoRef, newAddon.GetName())
+		if err != nil {
+			if strings.Contains(err.Error(), "directory not found") {
+				t.Logf("no need to upgrade test %s, it appears to be a new addon (no previous revisions found in branch %s)", newAddon.GetName(), comRepoRef)
+				addonDeploymentsArray = append(addonDeploymentsArray, newAddon)
+				continue
+			}
+			return false, nil, err
+		}
+		if oldAddon == nil {
+			t.Logf("no need to upgrade test %s, it appears to be a new addon (no previous revisions found in branch %s)", newAddon.GetName(), comRepoRef)
+			addonDeploymentsArray = append(addonDeploymentsArray, newAddon)
+			continue // new addon, upgrade test not needed
+		}
+
+		t.Logf("determining old and new versions for upgrade testing addon %s", newAddon.GetName())
+		oldRev := oldAddon.GetAnnotations()[constants.AddonRevisionAnnotation]
+		oldVersion, err := semver.Parse(strings.TrimPrefix(oldRev, "v"))
+		if err != nil {
+			return false, nil, err
+		}
+		newRev := newAddon.GetAnnotations()[constants.AddonRevisionAnnotation]
+		newVersion, err := semver.Parse(strings.TrimPrefix(newRev, "v"))
+		if err != nil {
+			return false, nil, err
+		}
+
+		if newVersion.EQ(oldVersion) {
+			t.Logf("skipping upgrade test for addon %s, it has not been updated", newAddon.GetName())
+			addonDeploymentsArray = append(addonDeploymentsArray, oldAddon)
+			continue
+		} else if oldVersion.GT(newVersion) {
+			return false, nil, fmt.Errorf("revisions for addon %s are broken, previous revision %s is newer than current %s", newAddon.GetName(), oldVersion, newVersion)
+		}
+
+		doUpgrade = true
```

Comment:
oops, sorry I meant the `else` wasn't needed to wrap the log but let's keep the log in! (https://github.com/mesosphere/kubernetes-base-addons/pull/809/commits/5da80d493f3257b938d403c8ad42bff58af846b7#diff-6b1d50a89d4ac455215d0f0969ed4e5507e3a91528e4ef6a20e37d6d515781b2L263-L264)
```suggestion
		t.Logf("found old version of addon %s %s (revision %s) and new version %s (revision %s)", newAddon.GetName(), oldRev, oldVersion, newVersion, newRev)
		doUpgrade = true
```